dspace-api/src/main/java/org/dspace/app/canvasdimension/IIIFCanvasDimensionServiceImpl.java

```diff
@@ -0,0 +1,219 @@
+        this.skipList = skipList;
+    }
+
+    @Override
+    public int processCommunity(Context context, Community community) throws Exception {
+        if (!inSkipList(community.getHandle())) {
+            List<Community> subcommunities = community.getSubcommunities();
+            for (Community subcommunity : subcommunities) {
+                processCommunity(context, subcommunity);
+            }
+            List<Collection> collections = community.getCollections();
+            for (Collection collection : collections) {
+                processCollection(context, collection);
+            }
+        }
+        return processed;
+    }
+
+    @Override
+    public int processCollection(Context context, Collection collection) throws Exception {
+        if (!inSkipList(collection.getHandle())) {
+            Iterator<Item> itemIterator = itemService.findAllByCollection(context, collection);
+            while (itemIterator.hasNext() && processed < max2Process) {
+                processItem(context, itemIterator.next());
+            }
+        }
+        return processed;
+    }
+
+    @Override
+    public void processItem(Context context, Item item) throws Exception {
+        if (!inSkipList(item.getHandle())) {
+            boolean isIIIFItem = IIIFSharedUtils.isIIIFItem(item);
+            if (isIIIFItem) {
+                if (processItemBundles(context, item)) {
+                    ++processed;
+                }
+            }
+        }
+    }
+
+    /**
+     * Process all IIIF bundles for an item.
+     * @param context
+     * @param item
+     * @return
+     * @throws Exception
+     */
+    private boolean processItemBundles(Context context, Item item) throws Exception {
+        List<Bundle> bundles = IIIFSharedUtils.getIIIFBundles(item);
+        boolean done = false;
+        for (Bundle bundle : bundles) {
+            List<Bitstream> bitstreams = bundle.getBitstreams();
+            for (Bitstream bit : bitstreams) {
+                done |= processBitstream(context, bit);
+            }
+        }
+        if (done) {
+            if (!isQuiet) {
+                System.out.println("Updated canvas metadata for item: " + item.getID());
+            }
+        }
+        return done;
+
+    }
+
+    /**
+     * Gets image height and width for the bitstream. For jp2 images, height and width are
+     * obtained from the IIIF image server. For other formats supported by ImageIO these values
+     * are read from the actual DSpace bitstream content. If bitstream width metadata already exists,
+     * the bitstream is processed when forceProcessing is true.
+     * @param context
+     * @param bitstream
+     * @return
+     * @throws Exception
+     */
+    private boolean processBitstream(Context context, Bitstream bitstream) throws Exception {
+        boolean processed = false;
+        boolean isUnsupported = bitstream.getFormat(context).getMIMEType().contains("image/jp2");
```

Comment:
it would be better imho to try to read any image and catch eventually the exceptions/null results